com.unity.netcode.gameobjects/Runtime/Messaging/MessagingSystem.cs

```diff
@@ -226,6 +259,66 @@ private bool CanReceive(ulong clientId, Type messageType, FastBufferReader messa
             return true;
         }
 
+        internal void ReorderMessage(int desiredOrder, uint targetHash)
+        {
+            if (desiredOrder < 0)
+            {
+                // invalid message re-ordering request. Ignore.
+                return;
+            }
+
+            if (desiredOrder < m_ReverseTypeMap.Length &&
+                XXHash.Hash32(m_ReverseTypeMap[desiredOrder].FullName) == targetHash)
+            {
+                // matching positions and hashes. All good.
+                return;
+            }
+
+            // Since the message at `desiredOrder` is not the expected one,
+            // insert an empty placeholder and move the messages down
+            var typesAsList = m_ReverseTypeMap.ToList();
+            typesAsList.Insert(desiredOrder, null);
+            m_ReverseTypeMap = typesAsList.ToArray();
+            var handlersAsList = m_MessageHandlers.ToList();
+            handlersAsList.Insert(desiredOrder, null);
+            m_MessageHandlers = handlersAsList.ToArray();
+
+            // we added a dummy message, bump the end up
+            m_HighMessageType++;
+
+            int position = desiredOrder;
+            bool found = false;
+            while (position < m_ReverseTypeMap.Length)
+            {
+                if (m_ReverseTypeMap[position] != null &&
+                    XXHash.Hash32(m_ReverseTypeMap[position].FullName) == targetHash)
+                {
+                    found = true;
+                    break;
+                }
+
+                position++;
+            }
+
+            if (found)
```

Comment:
Error if not found? Otherwise won't we have a mapping that maps to null?

Reply:
Yes. That is expected. Will comment further below. [1/2]

Reply:
Since this list is only synced one way (host->client), on the host this introduces the possibility of a null ref exception at `handler.Invoke(reader, ref context, this);` in `MessagingSystem.HandleMessage(...)` there's a catch there, but it's a generic one and the comment above it blames the user code. Can you add a specific check for a null handler?